sdk/storage/storage-blob/samples/typescript/proxyAuth.ts

```diff
@@ -17,7 +17,10 @@ async function main() {
     `https://${account}.blob.core.windows.net`,
     sharedKeyCredential,
     {
-      proxy: { url: "http://localhost:3128" }
+      proxy: {
+        // username and password can be passed as additional arguments
+        url: "http://localhost:3128"
```

Comment:
You may want to show actually passing a username and password

Reply:
Basic authentication works fine with current state of core-http if we pass in the `ProxySettings` as follows:
```javascript
  {
    host: "127.0.0.1",
    port: 3128,
    username: "xxx",
    password: "xxx"
  }
```
to the `proxyPolicy()` out of the box.

But if we wanted to load these values from the environment variables instead then we can have related changes in. Technically the loading can be done from SDK as well.

Reply:
@ramya0820 Good idea. I think `host` will also include `http://` or `https://`.

And [ProxyOptions](https://github.com/Azure/azure-sdk-for-js/blob/87a13978ccbc404aef035b92bf2a44d960fbc2d3/sdk/storage/storage-blob/src/Pipeline.ts#L70-L74) from storage sdks can be removed in favour of [ProxySettings](https://github.com/Azure/azure-sdk-for-js/blob/87a13978ccbc404aef035b92bf2a44d960fbc2d3/sdk/core/core-http/lib/serviceClient.ts#L38) from `core-http`.

We can directly pass `proxySettings` to the proxyPolicy() method and avoid using `getDefaultProxySettings`.
https://github.com/Azure/azure-sdk-for-js/blob/87a13978ccbc404aef035b92bf2a44d960fbc2d3/sdk/storage/storage-blob/src/Pipeline.ts#L224-L230


Thoughts, anybody ?

Reply:
Since the url to access is present on the request, believe core-http attempts to infer the protocol from it. 
Came across something about `username:password@host` being deprecated eventually.
Overall, if `ProxySettings` is exposed to user to pass in, that should be sufficient I think.
cc: @bterlson @chradek @ramya-rao-a

Reply:
Personal opinion here, I defer to @bterlson for the final call:

I would expect an sdk to use `getDefaultProxySettings` if the user has provided no details on the proxy and we are expected to infer from the environment.

If the user is indeed providing proxy details, then I would prefer them to pass in the `ProxySettings` instead of the proxy url (with host + port).

Reply:
Agree with @ramya-rao-a, we must load proxy settings from environment, so let's keep getDefaultProxySettings.

Reply:
ProxyOptions should not exist and be replaced with ProxySettings.

The maximally flexible signature would take either a ProxySettings or a string proxy URL which is parsed into a ProxySettings. Given the late hour, I'm ok with punting on parsing string proxy URLs for now.

Reply:
I would suggest something like https://github.com/ramya-rao-a/azure-sdk-for-js/commit/7dfbce17ea9a970c8386decc9b01c33bc3365fc2

Reply:
Another option is removing the options, and solely relying on the environment variables? It's Node-only anyway.